File: decompiled/3.17.1/be/ehealth/businessconnector/mycarenet/memberdata/builders/RequestObjectBuilderFactory.java
package be.ehealth.businessconnector.mycarenet.memberdata.builders;

import be.ehealth.technicalconnector.exception.TechnicalConnectorException;
import be.ehealth.technicalconnector.utils.ConfigurableFactoryHelper;

public final class RequestObjectBuilderFactory {
   private static final String PROP_ENCRYPTEDREQUESTBUILDER_CLASS = "memberdata.encryptedrequestobjectbuilder.class";
   private static final String PROP_NOTENCRYPTEDREQUESTBUILDER_CLASS = "memberdata.notencryptedrequestobjectbuilder.class";
   private static final String DEFAULT_ENCRYPTEDREQUESTBUILDER_CLASS = "be.ehealth.businessconnector.mycarenet.memberdata.builders.impl.EncryptedRequestObjectBuilderImpl";
   private static final String DEFAULT_NOTENCRYPTEDREQUESTBUILDER_CLASS = "be.ehealth.businessconnector.mycarenet.memberdata.builders.impl.NotEncryptedRequestObjectBuilderImpl";
   private static ConfigurableFactoryHelper<RequestObjectBuilder> helperFactoryERequestBuilder = new ConfigurableFactoryHelper("memberdata.encryptedrequestobjectbuilder.class", "be.ehealth.businessconnector.mycarenet.memberdata.builders.impl.EncryptedRequestObjectBuilderImpl");
   private static ConfigurableFactoryHelper<RequestObjectBuilder> helperFactoryNERequestBuilder = new ConfigurableFactoryHelper("memberdata.notencryptedrequestobjectbuilder.class", "be.ehealth.businessconnector.mycarenet.memberdata.builders.impl.NotEncryptedRequestObjectBuilderImpl");

   private RequestObjectBuilderFactory() {
   }

   public static RequestObjectBuilder getEncryptedRequestObjectBuilder() throws TechnicalConnectorException {
      return (RequestObjectBuilder)helperFactoryERequestBuilder.getImplementation();
   }

   public static RequestObjectBuilder getNotEncryptedRequestObjectBuilder() throws TechnicalConnectorException {
      return (RequestObjectBuilder)helperFactoryNERequestBuilder.getImplementation();
   }
}

File: src/main/kotlin/org/taktik/connector/business/memberdata/builders/RequestObjectBuilderFactory.kt
package org.taktik.connector.business.memberdata.builders

import org.taktik.connector.technical.exception.TechnicalConnectorException
import org.taktik.connector.technical.utils.ConfigurableFactoryHelper

object RequestObjectBuilderFactory {
    private val PROP_ENCRYPTEDREQUESTBUILDER_CLASS = "memberdata.encryptedrequestobjectbuilder.class"
    private val PROP_NOTENCRYPTEDREQUESTBUILDER_CLASS = "memberdata.notencryptedrequestobjectbuilder.class"
    private val DEFAULT_ENCRYPTEDREQUESTBUILDER_CLASS = "be.ehealth.businessconnector.mycarenet.memberdata.builders.impl.EncryptedRequestObjectBuilderImpl"
    private val DEFAULT_NOTENCRYPTEDREQUESTBUILDER_CLASS = "be.ehealth.businessconnector.mycarenet.memberdata.builders.impl.NotEncryptedRequestObjectBuilderImpl"
    private val helperFactoryERequestBuilder = ConfigurableFactoryHelper("memberdata.encryptedrequestobjectbuilder.class", "be.ehealth.businessconnector.mycarenet.memberdata.builders.impl.EncryptedRequestObjectBuilderImpl")
    private val helperFactoryNERequestBuilder = ConfigurableFactoryHelper("memberdata.notencryptedrequestobjectbuilder.class", "be.ehealth.businessconnector.mycarenet.memberdata.builders.impl.NotEncryptedRequestObjectBuilderImpl")

    val encryptedRequestObjectBuilder: RequestObjectBuilder
        @Throws(TechnicalConnectorException::class)
        get() = helperFactoryERequestBuilder.getImplementation()

    val notEncryptedRequestObjectBuilder: RequestObjectBuilder
        @Throws(TechnicalConnectorException::class)
        get() = helperFactoryNERequestBuilder.getImplementation()
}
